src/monitor.rs

```diff
@@ -69,48 +30,57 @@ impl fmt::Display for VideoMode {
 ///
 /// **Web:** A [`MonitorHandle`] created without
 #[cfg_attr(
-    any(web_platform, docsrs),
+    web_platform,
     doc = "[detailed monitor permissions][crate::platform::web::ActiveEventLoopExtWeb::request_detailed_monitor_permission]."
 )]
-#[cfg_attr(not(any(web_platform, docsrs)), doc = "detailed monitor permissions.")]
+#[cfg_attr(not(web_platform), doc = "detailed monitor permissions.")]
 /// will always represent the current monitor the browser window is in instead of a specific
 /// monitor. See
 #[cfg_attr(
-    any(web_platform, docsrs),
+    web_platform,
     doc = "[`MonitorHandleExtWeb::is_detailed()`][crate::platform::web::MonitorHandleExtWeb::is_detailed]"
 )]
-#[cfg_attr(not(any(web_platform, docsrs)), doc = "`MonitorHandleExtWeb::is_detailed()`")]
+#[cfg_attr(not(web_platform), doc = "`MonitorHandleExtWeb::is_detailed()`")]
 /// to check.
 ///
 /// [`Window`]: crate::window::Window
-#[derive(Clone, PartialEq, Eq, Hash, PartialOrd, Ord)]
-pub struct MonitorHandle {
-    pub(crate) inner: platform_impl::MonitorHandle,
+#[derive(Debug, Clone)]
+pub struct MonitorHandle(pub(crate) Arc<dyn MonitorHandleProvider>);
+
+impl Deref for MonitorHandle {
+    type Target = dyn MonitorHandleProvider;
+
+    fn deref(&self) -> &Self::Target {
+        self.0.as_ref()
+    }
 }
 
-impl std::fmt::Debug for MonitorHandle {
-    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        self.inner.fmt(f)
+impl PartialEq for MonitorHandle {
+    fn eq(&self, other: &Self) -> bool {
+        self.0.as_ref().eq(other.0.as_ref())
     }
 }
 
-impl MonitorHandle {
+impl Eq for MonitorHandle {}
+
+/// Provider of the [`MonitorHandle`].
+pub trait MonitorHandleProvider: AsAny + fmt::Debug + Send + Sync {
+    /// Native platform identifier of this monitor.
+    fn native_id(&self) -> u64;
```

Comment:
Researched a bit more, monitors are equatable on macOS, but the ID is a UUID (i.e. 128 bits), and comparing the naive ID (`CGDirectDisplayID`) is wrong (see https://github.com/rust-windowing/winit/blob/3a39a6ddb0927738cc97f6c0534d92c43f152728/src/platform_impl/apple/appkit/monitor.rs#L116-L118).

```suggestion
    fn native_id(&self) -> u128;
```